budget/app.py

```diff
@@ -106,8 +137,93 @@ def compare_budgets_and_users(account_id, users):
 
 def create_budget(synapse_id, team):
   '''Creates an AWS budget for a synapse user id'''
-  # TODO implement
-  log.info('Placeholder -- this will make a budget')
+  log.debug(f'Creating budget for synapse user {synapse_id}, member of team {team}')
+
+  team_budget_rules = TEAM_BUDGET_RULES.get('teams').get(team)
+  if not team_budget_rules:
+    raise ValueError(f'No budget rules available for team {team}')
+  budget_amount = team_budget_rules['amount']
+  budget_period = team_budget_rules['period']
+  budget = {
+    'BudgetName': f'{BUDGET_NAME_PREFIX}{synapse_id}',
+    'BudgetLimit': {
+      'Amount': budget_amount,
+      'Unit': 'USD'
```

Comment:
hmm, is your idea to put these other parameters (unit, IncludeRef, BudgetType, etc..) at the individual level so that it could be different for each team member?  if so how would it work if units are different for each member?  How do they get calculated towards the team budget?  

if that's not your intention then maybe these should be set in TEAM_BUDGET_RULES so it can be applied uniformly across all team members?

Reply:
The budget rules you are referring to are the budget rules for each user on a team. They are already applied uniformly across all team members. Later we will make the configuration a little more elaborate to allow overrides for individual users.  I am not following what you are confused about or whether you're suggesting something.

As for units, I actually debated whether to add this to the configuration at all, as we will always be using USD.

Reply:
@zaro0508 Yes, the configuration obtained through `TEAM_BUDGET_RULES['teams'][team_id]` where team_id is the id of a synapse team is applied to the budget created for a user. Budgets are created per user. This seems to be causing a lot of confusion. Do you have any suggestions on how this could be make more clear? I think it's going to have to be a code comment at the top of the file because I don't see a way to rename it that is going to fully clear it up.

Reply:
I think the way it's group in the json makes it confusing.  Items are grouped under teams (i.e. TEAM_BUDGET_RULES['teams']).  The amount seems like it should apply to the entire team not to individual users in a team.  How about we change the json and move the team inside of a rule, something like this..

```
USER_BUDGETS = {
  'rules': {
    'rule-100': {
      'amount': '10',
      'period': 'ANNUALLY',
      'community_manager_emails': ['tess.thyer@sagebase.org'],
      'team': '3412821'
    }
  },

```

Reply:
Where does the "100" come from? If you mean it should be driven by amount, there could be many teams, and some of them may allot the same amount per user.

Reply:
This problem of the confusion around the configuration name was discussed in a video chat and we agreed to add a comment explaining it a bit.